demos/common/cpp/models/src/classification_model.cpp

```diff
@@ -64,84 +69,110 @@ std::vector<std::string> ClassificationModel::loadLabels(const std::string& labe
-    if (inSizeVector.size() != 4 || inSizeVector[1] != 3)
-        throw std::runtime_error("3-channel 4-dimensional model's input is expected");
-    if (inSizeVector[2] != inSizeVector[3])
-        throw std::logic_error("Model input has incorrect image shape. Must be NxN square."
-                                " Got " + std::to_string(inSizeVector[2]) +
-                                "x" + std::to_string(inSizeVector[3]) + ".");
-
-    InferenceEngine::InputInfo& inputInfo = *cnnNetwork.getInputsInfo().begin()->second;
-    inputInfo.setPrecision(InferenceEngine::Precision::U8);
-    if (useAutoResize) {
-        inputInfo.getPreProcess().setResizeAlgorithm(InferenceEngine::ResizeAlgorithm::RESIZE_BILINEAR);
-        inputInfo.getInputData()->setLayout(InferenceEngine::Layout::NHWC);
+    // --------------------------- Prepare input  ------------------------------------------------------
+    if (model->inputs().size() != 1) {
+        throw std::logic_error("Classification model wrapper supports topologies with only 1 input");
     }
-    else
-        inputInfo.getInputData()->setLayout(InferenceEngine::Layout::NCHW);
-
-    // --------------------------- Prepare output blobs -----------------------------------------------------
-    const InferenceEngine::OutputsDataMap& outputsDataMap = cnnNetwork.getOutputsInfo();
-    if (outputsDataMap.size() != 1) throw std::runtime_error("Demo supports topologies only with 1 output");
-    InferenceEngine::Data& data = *outputsDataMap.begin()->second;
-    const InferenceEngine::SizeVector& outSizeVector = data.getTensorDesc().getDims();
-    if (outSizeVector.size() != 2 && outSizeVector.size() != 4)
-        throw std::runtime_error("Demo supports topologies only with 2-dimensional or 4-dimensional output");
-    if (outSizeVector.size() == 4 && (outSizeVector[2] != 1 || outSizeVector[3] != 1))
-        throw std::runtime_error("Demo supports topologies only with 4-dimensional output which has last two dimensions of size 1");
-    if (nTop > outSizeVector[1])
-        throw std::runtime_error("The model provides " + std::to_string(outSizeVector[1]) + " classes, but " + std::to_string(nTop) + " labels are requested to be predicted");
-    if (outSizeVector[1] == labels.size() + 1) {
-        labels.insert(labels.begin(), "other");
-        slog::warn << "\tInserted 'other' label as first." << slog::endl;
+    const auto& input = model->input();
+    inputsNames.push_back(input.get_any_name());
+
+    const ov::Shape& inputShape = input.get_shape();
+    ov::Layout inputLayout;
+    if (!layouts.empty()) {
+        inputLayout = layouts.begin()->second;
+    }
+    else {
+        inputLayout = getLayoutFromShape(inputShape);
+        slog::warn << "Layout for input \"" << input.get_any_name() << "\" was not set explicitly. "
+            << "Automatically detected layout \"" << inputLayout.to_string() << "\" will be used." << slog::endl;
```

Comment:
That depends. If all models in OMZ for the same usecase will have same layout after conversion - OK, we can use defaults/heuristics. Otherwise, if, for instance TF-based models will keep NHWC after conversion, and others (used with the same wrapper) will keep NCHW - we have to emphasize  that -layout better should be set explicitly.
I'd leave this message as is, as a reminder.

Reply:
@fzhar by the time you wrote this, I've already removed it)